server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -321,6 +328,64 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.cachedTimeThread.start();
     }
 
+    private static HashMap<String, LongGauge> setupMetrics(MeterRegistry meterRegistry, String name, ExecutorHolder holder) {
+        Map<String, Object> at = null;
+        HashMap<String, LongGauge> gauges = new HashMap<>();
+        if (holder.executor() instanceof ThreadPoolExecutor threadPoolExecutor) {
+            String prefix = "es.tp." + name + ".t."; // temporarily shortening due to max 68 chars limit
```

Comment:
Yes, we hit the limit already - the full name i was thinking was es.thread_pool.<name>.thread.<metric>.
(i see the limit was 63, not 68 like I wrote)
Some of the threadpools have fairly long names. E.g » 
[2023-11-28T10:23:04,176][ERROR][o.e.b.Elasticsearch      ] [runTask-0] fatal exception while booting Elasticsearch java.lang.IllegalArgumentException: Instrument name [es.thread_pool.searchable_snapshots_cache_fetch_async.threads.count] with length [67] exceeds maximum length [63]

I know it will be disruptive to change afterwards, I am not sure what sort of timeline there is for the new apm agent, if thats not too far away we could perhaps wait on merging until that's available.

Reply:
I should have this fixed soon https://github.com/elastic/elasticsearch/pull/102691

Reply:
https://github.com/elastic/elasticsearch/pull/102691 is now merged, @gareth-ellis you can update the names and use up to 255 characters :)

Reply:
Fixed now in  fc45e177b705d1b49fff5039f1cafd714c267d97